### DISCUSSION TITLE
How to use Python

### DISCUSSION THREAD

**Discussion by zl530380** [utelle/wxsqlite3#118] [Q&A]
How to use wxSqLite3 in Python
![20241209-152831](https://github.com/user-attachments/assets/cbeaec19-f08f-447f-b59c-9aa0a3b6303a)

**Comment by utelle** [comment]
> How to use wxSqLite3 in Python

As far as I can tell you are showing a screenshot from the SQLite database management tool [SQLiteStudio](https://github.com/pawelsalawa/sqlitestudio). The selection **WxSQLite3** for the _database type_ is somewhat misleading, because under the hood _SQLiteStudio_ uses [SQLite3 Multiple Ciphers](https://github.com/utelle/SQLite3MultipleCiphers) - on which [wxSQLite3](https://github.com/utelle/wxsqlite3) is based nowadays, too.

For accessing databases encrypted with [SQLite3 Multiple Ciphers](https://github.com/utelle/SQLite3MultipleCiphers) from _**Python**_ you can use the _Python package_ [apsw-sqlite3mc](https://github.com/utelle/apsw-sqlite3mc) which is hosted on [PyPI](https://pypi.org/project/apsw-sqlite3mc/) and can be installed in _**Python**_ with `pip`:

```Python
pip install apsw-sqlite3mc
```

Since this is actually not an issue with _**wxSQLite3**_ I will convert the issue into a discussion.

**Comment by sorrowcat1** [comment]
> 如何在 Python 中使用 wxSqLite3 ![20241209-152831](https://private-user-images.githubusercontent.com/16401564/393762731-cbeaec19-f08f-447f-b59c-9aa0a3b6303a.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjYzOTkyNzUsIm5iZiI6MTc2NjM5ODk3NSwicGF0aCI6Ii8xNjQwMTU2NC8zOTM3NjI3MzEtY2JlYWVjMTktZjA4Zi00NDdmLWI1OWMtOWFhMGEzYjYzMDNhLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjIyVDEwMjI1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjZTViYTZlNGY4MDBiYjZlNGMzMDkyMmJhNWUxNDM1ZjRlM2ZhODg1ZTc3NzM2MTNjZDBjYTJiYzg5NjM4MGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nTYKcm5-ZnHY4bky3t4Snv2JCSiyrDbmJyLA-iydoUA)


Has this problem been resolved?
